chef/automate
# Associate project for every organisation fetched

### :nut_and_bolt: Description: What code changed, and why?
When a organisation is migrated from the chef server to automate, an IAM project needs to be linked with it. 
So added the call to project create service to create the project with the name of serverId_orgID, every time the organisation is stored in the database the project is created with the name having all the policies as editors, viewer, owner.

### :chains: Related Resources



### :+1: Definition of Done
Project is created and stored in database at the time of migration

### :athletic_shoe: How to Build and Test the Change
Steps to build:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
Steps to test:

1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
2. Add the chef-server test environment credentials and extract the SERVER IDs

Request
This is a get request.
```curl -sSkH "api-token: $(get_admin_token)" 'https://a2-dev.test/api/v0/infra/servers/SERVER_ID/infraserverorgs'```

Response

```
{
    "orgs": [
        {
            "id": "test",
            "name": "test",
            "admin_user": "",
            "credential_id": "",
            "server_id": "serverId",
            "projects": ["serverID_test"]
        }
    ]
}
```

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
